parmed/structure.py

```diff
@@ -1689,17 +1689,41 @@ def topology(self):
                     last_chain = atom.residue.chain
                     chain = top.addChain(id=last_chain)
                 last_residue = atom.residue
-                last_omm_residue = top.addResidue(atom.residue.name, chain)
+                last_omm_residue = top.addResidue(
+                    atom.residue.name,
+                    chain,
+                    id=str(atom.residue.number),
+                    insertionCode=atom.residue.insertion_code,
+                )
             try:
                 elem = app.element.Element.getByAtomicNumber(
                     atom.atomic_number)
             except KeyError:
                 elem = None
             top.addAtom(atom.name, elem, last_omm_residue)
         # Add the bonds
+        bond_orders_by_openmm_type = {
+            app.Single: 1.0,
+            app.Double: 2.0,
+            app.Triple: 3.0,
+            app.Amide: 1.25,
+            app.Aromatic: 1.5,
+        }
         atoms = list(top.atoms())
         for bond in self.bonds:
-            top.addBond(atoms[bond.atom1.idx], atoms[bond.atom2.idx])
+            # Determine the bond type
+            type = None
+            for t, o in bond_orders_by_openmm_type.items():
+                if math.isclose(bond.order, o):
+                    type = t
+            # Determine the bond order (an integer)
+            order = None
+            n = round(bond.order)
+            if math.isclose(bond.order, n):
+                order = n
+            top.addBond(
+                atoms[bond.atom1.idx], atoms[bond.atom2.idx], type=type, order=order
+            )
```

Comment:
I've also recently added a `qualitative_type` that is basically a translation of the RDKit bond types.  Not sure if that would be useful here or not.

Reply:
Hmm... maybe. Would it make sense to add
```py
qualitative_type = None
if b.type is not None:
    name = repr(b.type).upper()
    if hasattr(QualitativeBondType, name):
        qualitative_type = getattr(QualitativeBondType, name)
```
to `parmed.openmm.load_topology()`, and do something like
```py
type = None
if bond.qualitative_type is not None:
    name = bond.qualitative_type.name.lower().capitalize()
    if hasattr(app, name):
        t = getattr(app, name)
        if t in bond_orders_by_openmm_type:  # ensure t is a recognized type
            type = t
else:  # try determining the type from the bond order
    for t, o in bond_orders_by_openmm_type.items():
        if math.isclose(bond.order, o):
            type = t
```
in `Structure.topology.getter()`?

Reply:
I'm tempted to say let's hold off on that until and unless it proves useful, and let's just get this change merged.